mlflow/models/evaluation/default_evaluator.py

```diff
@@ -1447,7 +1447,17 @@ def _check_args(self, metrics, eval_df):
             output_columns = (
                 [] if self.other_output_columns is None else list(self.other_output_columns.columns)
             )
+            if self.dataset.predictions_name:
+                output_columns.append(self.dataset.predictions_name)
+            else:
+                output_columns.append("predictions")
```

Comment:
Is there a reason we are hardcoding this to predictions and not `self.predictions` which is the actual column_name passed to predictions?

Reply:
I guess I was basing this off of the logic in `_get_args_for_metrics()` - it checks for column == "predictions" (hardcoded) or columns == self.dataset.predictions_name. Not sure what I'm missing here? I was under the impression self.dataset.predictions_name is the column_name passed to predictions

Reply:
> I was under the impression self.dataset.predictions_name is the column_name passed to predictions

Thats only the case when the predictions column is passed via the dataset. If we use model.predict then that is not the case. Check this PR for the bug fix I did: https://github.com/mlflow/mlflow/pull/10160